internal/server/logbuffer/merge.go

```diff
@@ -0,0 +1,153 @@
+package logbuffer
+
+import (
+	"io"
+	"time"
+)
+
+// LogMerge can combine multiple log streams into one stream. It presumes
+// each stream emits entries in time order, and then weaves the entries
+// together to create a total time ordered stream.
+type LogMerge struct {
+	inputs []LogMergeInput
+	heads  []TimedEntry
+}
+
+// TimedEntry is the interface each input returns entries in
+type TimedEntry interface {
+	// Time is the associated time of the value. This is used to sort
+	// the entry against other entries from other inputs.
+	Time() time.Time
+
+	// Value returns the log entry itself. This allows time addition
+	// wrappers to return their original value.
+	Value() interface{}
+}
+
+// LogMergeInput is value that returns TimedEntry's for LogMerge
+// to weave together.
+type LogMergeInput interface {
```

Comment:
Going along with above, bikeshed but perhaps `MergeReader` makes more sense here, so you're composing Reader types similar to `io` and `Reader` happens to also implement `MergeReader`.

Reply:
Totes agree!